extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -715,6 +715,25 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     return {'artc': artc.join() || null, 'ati': ati.join(), 'ard': ard.join()};
   }
 
+  /** @override */
+  getCustomRealTimeConfigMacros_() {
+    const whitelist = {
+      'height': true,
+      'width': true,
+      'data-slot': true,
+      'data-multi-size': true,
+      'data-multi-size-validation': true,
+      'data-override-width': true,
+      'data-override-height': true,
+      'data-json': true,
+    };
+    return {
+      PAGEVIEWID: () => Services.documentInfoForDoc(this.element).pageViewId,
+      HREF: () => this.win.location.href,
+      ATTR: name => !!whitelist[name] && this.element.getAttribute(name),
```

Comment:
You could be a bit fancier and dev or user warn on attempted access to missing attribute.  Also, do you know if element names are case sensitive?  Perhaps this should be insensitive?

Reply:
done